### ISSUE TITLE
Tauren Hitbox Racial

### ISSUE THREAD

**Issue by SuddenDeath** [LightsHope/issues#332]
The Problem: Taurens should have significantly larger hitbox than all other races

How it should work:
Taurens should have hitbox/combatreach of 4.05 yards.

Basically all player races should have similar or the same hitbox as their NPC/Mob counterparts in Vanilla, as far as I know.. All except taurens hitboxes are 1.5 yards.

Hitboxes will change depending on if they are in a form that changes their model(druid forms, ghostwolf etc)

Proof:

Vanilla video: https://www.youtube.com/watch?v=UaHRC8PJkeE (he turned himself into a tauren with orb of deception which gives him tauren hitbox for the duration of the effect)
@ 2:50 and 4:09.

Patch 2.4.2 video: https://www.youtube.com/watch?v=xjMRzrCOo2w
Source: http://www.warcraftmovies.com/movieview.php?id=82055

(The Tauren warr vs Tauren druid in the video obviously displays this quite a lot since they are both taurens it results in an almost double up situation, which explains the obnoxious long range)

https://youtu.be/oo_vcsdBKXc?t=12m
-doesn't get very close with Intercept(which is because of this too), yet still able to hit the druid right away

https://youtu.be/4JnQaXSHzEE?t=12m20s
https://youtu.be/oo_vcsdBKXc?t=7m50s

Some old forum threads about it:
http://www.dual-boxing.com/archive/index.php/t-7338.html (Also mentions cat form as tauren druid)
http://www.arenajunkies.com/topic/19398-tauren-swing-range-all/

Different hitbox in forms:
http://i.imgur.com/FDno7tU.gif (out of range when in travel form but not tauren)
Source: https://youtu.be/Rkb32GWDEzE?t=30m50s

How I determined approximately 4.0 value(see additional info at the bottom of report):
https://www.youtube.com/watch?v=geqf5cvuP7E
(9.18 - 1.33) / 2 = 3.925

Also if you check the hitbox of tauren NPC/Mobs ingame on retail, they seem to have about 4.0 as well:
https://youtu.be/5FvqqB_p8h4 - Still the same with tauren mobs on retail

For this to work, there needs to be proper melee range calculation in place.

How it's calculated in the client:
range_bonus = m_caster->GetCombatReach() + 1.33f + (target ? target->GetCombatReach() : m_caster->GetCombatReach());

Basically:
Melee range = max(5,combatreach_player + combatreach_target + 1.33)

EDIT: Additional proof from /u/killerwife on reddit(discussing sniffed data from retail TBC)
Okay so. My investigation found the following: Client has default combat reach stored SOMEWHERE (its not sent on initial packet send). The way I had to track it down, was to find a tauren which shapeshifts into druid form and then resets.

This is the result: [0] (150) UNIT_FIELD_BOUNDINGRADIUS: 0.9747 [0] (151) UNIT_FIELD_COMBATREACH: 4.05 [0] (152) UNIT_FIELD_DISPLAYID: 59

**Comment by ratkosrb** [comment]
Currently it is set to not update combat reach on model change for players. It always uses 1.5.

https://github.com/lh-server/core/blob/development/src/game/Objects/Unit.cpp#L9924

**Comment by slk333** [comment]
this is an important issue, I suggest fixing this asap

**Comment by sadfx** [comment]
This isn't simply an issue of combat reach, your size should be changed to the new model as well. Transforming from Race A to Race B still leaves you with the size of Race A.

**Comment by slk333** [comment]
up
